x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/total_feature_importane_summary/feature_importance_summary.tsx

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useCallback, useMemo } from 'react';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIconTip,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  Chart,
+  Settings,
+  Axis,
+  ScaleType,
+  Position,
+  BarSeries,
+  RecursivePartial,
+  AxisStyle,
+  PartialTheme,
+  BarSeriesSpec,
+} from '@elastic/charts';
+import { i18n } from '@kbn/i18n';
+import euiVars from '@elastic/eui/dist/eui_theme_light.json';
+import {
+  TotalFeatureImportance,
+  isClassificationTotalFeatureImportance,
+  isRegressionTotalFeatureImportance,
+  RegressionTotalFeatureImportance,
+  ClassificationTotalFeatureImportance,
+} from '../../../../../../../common/types/inference';
+import { useMlKibana } from '../../../../../contexts/kibana';
+const { euiColorMediumShade } = euiVars;
+const axisColor = euiColorMediumShade;
+
+const axes: RecursivePartial<AxisStyle> = {
+  axisLine: {
+    stroke: axisColor,
+  },
+  tickLabel: {
+    fontSize: 12,
+    fill: axisColor,
+  },
+  tickLine: {
+    stroke: axisColor,
+  },
+  gridLine: {
+    horizontal: {
+      dash: [1, 2],
+    },
+    vertical: {
+      strokeWidth: 0,
+    },
+  },
+};
+const theme: PartialTheme = {
+  axes,
+};
+
+export interface FeatureImportanceSummaryPanelProps {
+  totalFeatureImportance: TotalFeatureImportance[];
+}
+
+const tooltipContent = i18n.translate(
+  'xpack.ml.dataframe.analytics.exploration.featureImportanceSummaryTooltipContent',
+  {
+    defaultMessage:
+      'Shows to what degree a given feature of a data point contributes to the prediction. The magnitude of feature importance shows how significantly the feature affects the prediction both locally (for a given data point) or generally (for the whole data set).',
+  }
+);
+
+export const FeatureImportanceSummaryPanel: FC<FeatureImportanceSummaryPanelProps> = ({
+  totalFeatureImportance,
+}) => {
+  const {
+    services: { docLinks },
+  } = useMlKibana();
+
+  const [plotData, barSeriesSpec] = useMemo(() => {
+    let sortedData: Array<{
+      featureName: string;
+      meanImportance: number;
+      className?: string;
+    }> = [];
+    let barSettings: Partial<BarSeriesSpec> = {
+      xAccessor: 'featureName',
+      yAccessors: ['meanImportance'],
+    };
+    if (totalFeatureImportance.length < 1) {
+      return [sortedData, barSettings];
+    }
+
+    if (isClassificationTotalFeatureImportance(totalFeatureImportance[0])) {
+      // if binary classification
+      if (totalFeatureImportance[0].classes.length === 2) {
+        sortedData = (totalFeatureImportance as ClassificationTotalFeatureImportance[]).map((d) => {
```

Comment:
Can this be updated to remove the `as ClassificationTotalFeatureImportance`? Maybe setting  `totalFeatureImportance[0]` to it's own const and then the `isClassificationTotalFeatureImportance` type guard would be enough?